src/material/input/input.ts

```diff
@@ -332,6 +333,16 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
     // FormsModule or ReactiveFormsModule, because Angular forms also listens to input events.
   }
 
+  /** Determines the value of the native `placeholder` attribute that should be used in the DOM. */
+  _getNativePlaceholderValue() {
+    // If we're hiding the native placeholder, it should also be cleared from the DOM, otherwise
+    // screen readers will read it out twice: once from the label and once from the attribute.
+    // TODO: can be removed once we get rid of the `legacy` style for the form field, because it's
+    // the only one that supports promoting the placeholder to a label.
+    const formField = this._formField;
+    return (!formField || !formField._hideControlPlaceholder()) ? this.placeholder : null;
```

Comment:
Small issue found in testing that needs to be resolved before we can merge.

This function calls the form field's `_hideControlPlaceholder()`, which checks the control's `_shouldLabelFloat`. In the case where the control is `MatChipList`, that property is a getter:

```ts
  get shouldLabelFloat(): boolean { return !this.empty || this.focused; }
```

It checks the `empty` property of the `MatChipList`, which is also a getter:

```ts
  get empty(): boolean {
    return (!this._chipInput || this._chipInput.empty) && this.chips.length === 0;
  }
```

Here we find that `this.chips.length` throws an exception because `this.chips` is `undefined` since the content of the list has not yet been initialized.

Here's a reproduction of the issue we are seeing: https://stackblitz.com/edit/components-issue-2obaho?file=app%2Fapp.component.ts

It's using the same hook timing of the placeholder host binding, and you can see in the debug console that an exception is thrown when calling `_hideControlPlaceholder()`